test/generators/component_generator_test.rb

```diff
@@ -13,42 +13,50 @@ class ComponentGeneratorTest < Rails::Generators::TestCase
 
   arguments %w[user]
 
-  def test_component_with_required_content
-    Thor::LineEditor.stub :readline, "Y" do
-      run_generator
-    end
+  def test_component_without_required_content
+    without_required_content { run_generator }
 
     assert_file "app/components/user_component.rb" do |component|
       assert_match(/class UserComponent < /, component)
-      assert_match(/validates :content, presence: true/, component)
+      refute_match(/validates :content, presence: true/, component)
     end
 
     assert_file "app/components/user_component.html.erb" do |view|
-      assert_match(/<%= content %>/, view)
+      assert_match(/<div>Add User template here<\/div>/, view)
     end
   end
 
-  def test_component_without_required_content
-    Thor::LineEditor.stub :readline, "n" do
-      run_generator
-    end
+  def test_component_with_required_content
+    with_required_content { run_generator }
```

Comment:
why reverse the order of these tests. It makes this diff super hard to follow.

Reply:
Good point and I'm not sure why I did this 😬 Added a new commit, should be easier to follow now.